plugins/s3_auth/Makefile.inc

```diff
@@ -15,4 +15,4 @@
 #  limitations under the License.
 
 pkglib_LTLIBRARIES += s3_auth/s3_auth.la
-s3_auth_s3_auth_la_SOURCES = s3_auth/s3_auth.cc
+s3_auth_s3_auth_la_SOURCES = s3_auth/s3_auth.cc s3_auth/aws_auth_v4.cc
```

Comment:
Initially I named it s3_auth_v4 but realized that there is nothing s3 specific (except the entry-point to region map, which can be overridden from a file) and that is why I renamed it to "aws_auth_v4" (it can be used with any AWS service)